frontend/src/pages/GamePage.tsx

```diff
@@ -1,99 +1,200 @@
+        const data = await response.json();
 
-        // Update terminal lines
-        setTerminalLineData((prevData) => [
+        if (data.length > 0) {
+          setQuestions(data);
+          const firstQuestionText = `Question 1: ${data[0].title}`;
+          const firstOptionsText = data[0].options
+            .map((option: Option, index: number) => `${index + 1}. ${option.text}`)
+            .join("\n");
+          setTerminalLineData((prevData) => [
             ...prevData,
-            <br />,
-            <TerminalOutput>{`You chose: ${selectedOption.text}`}</TerminalOutput>,
-            <br />,
-            <TerminalOutput>
-                {`Updated stats - Health: ${stats.health + selectedOption.impact.health}, Happiness: ${
-                    stats.happiness + selectedOption.impact.happiness
-                }, Money: ${stats.money + selectedOption.impact.money}.`}
-            </TerminalOutput>,
-            <br />,
-        ]);
-
-        // Move to the next question
-        const nextQuestionIndex = currentQuestionIndex + 1;
-        if (nextQuestionIndex < quizQuestions.length) {
-            setCurrentQuestionIndex(nextQuestionIndex);
-            const nextQuestion = quizQuestions[nextQuestionIndex];
-            setTerminalLineData((prevData) => [
-                ...prevData,
-                <br />,
-                <TerminalOutput>{`Question ${nextQuestionIndex + 1}: ${nextQuestion.question}`}</TerminalOutput>,
-                ...nextQuestion.options.map((option, index) => (
-                    <TerminalOutput key={`t-${index}`}>{`${index + 1}. ${option.text}`}</TerminalOutput>
-                )),
-            ]);
+            firstQuestionText,
+            firstOptionsText,
+          ]);
         } else {
-            setTerminalLineData((prevData) => [
-                ...prevData,
-                <TerminalOutput>
-                    {`Game Over! Final stats - Health: ${stats.health}, Happiness: ${stats.happiness}, Money: ${stats.money}.`}
-                </TerminalOutput>,
-            ]);
+          setGameOver(true);
+          setTerminalLineData((prevData) => [...prevData, "No questions available. Game over!"]);
         }
+      } catch (err) {
+        setError(err instanceof Error ? err.message : "An error occurred.");
+      } finally {
+        setIsLoading(false);
+      }
     };
 
-    return (
-        <div className="container">
-            <Terminal
-                name="Life Choices Game"
-                colorMode={ColorMode.Dark}
-                onInput={handleInput}
-            >
-                {terminalLineData}
-            </Terminal>
-        </div>
+    fetchQuestions();
+  }, []);
+
+  const submitAnswer = async (
+    questionId: string,
+    selectedOption: string,
+    newStats: { health: number; wealth: number; happiness: number }
+  ): Promise<boolean> => {
+    try {
+      const response = await fetch("https://kernel-panic.onrender.com/api/submitAnswer", {
```

Comment:
Why are you still using fetch api even though I have a better instance created as `PanicApi`.
Replace this by `PanicApi.post("/submitAnswer")`

Reply:
For info, axios provides many benefits over the fetch api and to have a good code structure, reuse the instance I created